policy-test/tests/outbound_api.rs

```diff
@@ -158,7 +162,16 @@ async fn service_with_http_routes_with_backend() {
         let backends = [backend_name];
         let _route = create(
             &client,
-            mk_http_route(&ns, "foo-route", &svc, 4191, Some(&backends), None),
+            mk_http_route(
+                &ns,
+                "foo-route",
+                &svc,
+                4191,
+                Some(&backends),
+                None,
+                None,
+                None,
+            ),
```

Comment:
nit, not actually important: this feels like enough arguments that it *might* be worth adding something like
```rust
struct NewHttpRoute<'a> {
    ns: &'a str,
    name: &'a str,
    svc: &'a k8s::Service,
    port: u16,
    backends: Option<&'a [&'a str]>,
    backends_ns: Option<String>,
    filters: Option<Vec<k8s::policy::httproute::HttpRouteFilter>>,
    backend_filters: Option<Vec<k8s_gateway_api::HttpRouteFilter>>,
}
```
and changing `mk_http_route` to a method on that struct or something. That way, we would be naming all these fields at the callsite, rather than just having a big pile of `None`s and having to look at the function definition to know what they mean.

that would also make the [clippy lint for "too many arguments"](https://github.com/linkerd/linkerd2/pull/11083/files#diff-8564fea6da1b70c1515b4d4aaf8497ebbea443c221d889d102d3e486c8f8917aR930) go away.

Reply:
replaced with a builder pattern